src/Microsoft.DotNet.CodeFormatting/ResponseFileWorkspace.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.DotNet.CodeFormatting
+{
+    public sealed class ResponseFileWorkspace : Workspace
+    {
+        private ResponseFileWorkspace()
+            : base(DesktopMefHostServices.DefaultServices, "Custom")
+        {
+            LoadMetadataForReferencedProjects = true;
+        }
+
+        public static ResponseFileWorkspace Create()
+        {
+            return new ResponseFileWorkspace();
+        }
+
+        public Project OpenCommandLineProject(string responseFile, string language)
+        {
+            // This line deserves better error handling, but the tools current model is just throwing exception for most errors.
+            // Issue: #90
+            string rspContents = File.ReadAllText(responseFile);
+
+            var projectInfo = CommandLineProject.CreateProjectInfo(
+                projectName: Path.GetFileNameWithoutExtension(responseFile),
+                language: language,
+                commandLine: rspContents,
+                baseDirectory: Path.GetDirectoryName(Path.GetFullPath(responseFile)),
+                workspace: this);
+
+            this.OnProjectAdded(projectInfo);
+
+            return this.CurrentSolution.GetProject(projectInfo.Id);
+        }
+
+        public override bool CanApplyChange(ApplyChangesKind feature)
+        {
+            return feature == ApplyChangesKind.ChangeDocument;
+        }
+
+        protected override void ApplyDocumentTextChanged(DocumentId documentId, SourceText text)
+        {
+            var document = this.CurrentSolution.GetDocument(documentId);
+            if (document != null)
+            {
+                try
+                {
+                    var filePath = document.FilePath;
+                    Directory.CreateDirectory(Path.GetDirectoryName(filePath));
+                    using (var writer = new StreamWriter(filePath, append: false, encoding: text.Encoding ?? Encoding.UTF8))
```

Comment:
Done, but you should probably file a bug on Roslyn because the MsBuildWorkspace emits with BOM. Other side-note: I've seen some of the git tools infer the wrong encoding if the BOM isn't there. Causing big issues with the windows vs Unix line endings mess.